### ISSUE TITLE
Analysis & Equipment

### ISSUE THREAD

**Issue by uk3** [Runalyze/Runalyze#1503]
Hi,

An Enhancement-Idea: In the Analysis Tab it would be useful to also have the analysis split 'per Equipment'

e.g. when biking: How many kms, hours did I use which bike (analoge to the analysis per Training Type

lG Ulrich

**Comment by laufhannes** [comment]
I guess you mean what was in the 'weather' plugin for clothes before our equipment refactoring, right?

I'm not sure if it fits in the 'analysis' plugin well, as that tab is already quite big. But probably it's too much duplicate code if we put it into a new plugin. Maybe it'd be good to add another dropdown:
 - 'Zones & Types'
 - 'Equipment'
 - probably more*

* e.g. 'distance zones' (< 5k, 5 to 10k, ...), 'time zones' (< 30min, 30 to 60min, ...), ...

**Comment by uk3** [comment]
Hello,

Yes, is the same topic i guess.

I saw it that way, because before the equipment refactoring I tracked the Bicycles as Training Types. This then was reflected in the Analysis plugin.

But you are Probably right, an other drop-down would be a cleaner and more flexible solution. 

lG uk

> On 02 Oct 2015, at 18:18, Hannes <notifications@github.com> wrote:
> 
> I guess you mean what was in the 'weather' plugin for clothes before our equipment refactoring, right?
> 
> I'm not sure if it fits in the 'analysis' plugin well, as that tab is already quite big. But probably it's too much duplicate code if we put it into a new plugin. Maybe it'd be good to add another dropdown:
> 
> 'Zones & Types'
> 'Equipment'
> probably more*
> e.g. 'distance zones' (< 5k, 5 to 10k, ...), 'time zones' (< 30min, 30 to 60min, ...), ...
> —
> Reply to this email directly or view it on GitHub <https://github.com/Runalyze/Runalyze/issues/1503#issuecomment-145075982>.
> 

-- 
Ulrich Kiermayr, Vienna, AT
mail & jabber: ulrich@kiermayr.at <mailto:ulrich@kiermayr.at>

**Comment by grzeg1** [comment]
Isn't "show all equipment" table opened straight from plugin enough for this purpose?

**Comment by uk3** [comment]
Hello,

> On 02 Oct 2015, at 18:24, grzeg1 <notifications@github.com> wrote:
> 
> Isn't "show all equipment" table opened straight from plugin enough for this purpose?
> 

I am Interested (esp. regarding Bikes) on a breakdown on a per month / per year basis. (like how many kms did I do on my Roadbike this month).

The show all equipment does not reflect this.


lG uk
> —
> Reply to this email directly or view it on GitHub <https://github.com/Runalyze/Runalyze/issues/1503#issuecomment-145078442>.
> 

-- 
Ulrich Kiermayr, Vienna, AT
mail & jabber: ulrich@kiermayr.at <mailto:ulrich@kiermayr.at>

**Comment by grzeg1** [comment]
True, it does not. But for Road/MTB you still have different training types.

The hard thing is that with more filters it gets more and more complicated. @laufhannes gave a good example above. A few days ago I was thinking about having separate VDOT charts/prognoses for short and long runs (short are limited by aerobic capacity, long by muscle fatigue) to track progress. I was also thinking about separating flat and hilly/trail runs.
Maybe we should re-use workout search window to serve as a filter for analyses. It would return a list of training ids that would then be injected into WHERE clauses? Then we'd have full freedom.
To allow easier usage we could allow user to save filters with names and call them up when needed.

**Comment by uk3** [comment]
Hello,

i Unverstand, but esp. For bikes training type (e.g. tour, intervals,
hills, etc.) and type of bike are 2 different things for the same activity.
If I still have to track the bikes in different training types or even
sports, it makes the equipment tracking not very useful.

Lg Ulrich

**Comment by grzeg1** [comment]
I always thought about equipment tracking in the view of gear replacement/servicing.
Checking how long shoes will last and when I'll need a new pair or when the bike's chain will need to be replaced.

**Comment by laufhannes** [comment]
I understand that 'show all equipment' is not the same. For running shoes there's probably no need to analyze which shoe was used in which month and so on, but e.g. for clothes it's interesting. And, anyway, as there are much more groups that could be analyzed (flat vs. hilly is one example), it's no problem to add this feature for equipment as well.

**Selection criteria**: I'm not sure if it's good to use the search window for that. There will be more plugins that could use user-specified search conditions. I like the idea of having a dynamic form within the plugin itself where conditions can be added dynamically.

**More filter/groups**: Adding a dynamic form for that would be the most flexible and powerful solution (but probably the most complicated one as well). Users may want to use different groups, e.g. different definitions of flat vs. hilly (in terms of avg. gradient).

**Comment by mipapo** [comment]
Some additional ideas are mentioned here #1740
